### PR TITLE
feat: Handle more expressive types++

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
- fixes https://github.com/ipvm-wg/ipvm/issues/24
- Ipld <=> Wit/Wasmtime Value Types
- Wasmtime shim for instantiation, execution, parsing component wasm mods
- Allows for Wats if given Wasm-component Wats
- Store in database via to/from sql functions and necessary types
- More robust receipt implementation across pubsub/wire
- Moves project to workspace
- Adds ipvm-guest-wasm package for testing/compiling out wasms w/ Wit IDLs
- Fixtures for testing variations
- Cleanup

### CODE REVIEW

**Comment by zeeshanlakhani** [comment]
FYI, cc @expede and others, nightly linting fails currently due to https://github.com/rust-lang/rust-clippy/issues/10318.

File: ipvm/src/network/client.rs [hunk 1]
```diff
@@ -110,43 +113,70 @@ impl Client {
...
     RespondFile {
+        /// File content.
         file: Vec<u8>,
+        /// Channel to send on.
         channel: ResponseChannel<FileResponse>,
     },
+    /// Publish message on the topic name.
     PublishMessage {
+        /// Pubsub (i.e. [libp2p::gossipsub]) topic.
         topic: String,
```
Comment (expede) [inline ipvm/src/network/client.rs:176]: Super minor thing, and possibly Haskell-brain happening, but would it be helpful to track that this is not just any string with a newtype `Topic`?

Comment (zeeshanlakhani) [reply ipvm/src/network/client.rs:176]: Yeah, a newtype may work. I won't touch `File` and such yet b/c I think the whole networking layer needs a nice refactor anyhow. It's still a lot of the example code tbh (w/ additions).

**Comment by zeeshanlakhani** [comment]
@expede merging as per our discussion yesterday.

**Pull request merged**